### PR TITLE
CB-17306 Add consumption-api checks to CI swagger-compatibility-test

### PR DESCRIPTION
**Pull request by doktoric**
See detailed description in the commit message.

### CODE REVIEW

File: cloud-consumption/src/test/java/com/sequenceiq/consumption/swagger/SwaggerGenerator.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package com.sequenceiq.redbeams.swagger;
```
Comment (lajosrodek) [inline cloud-consumption/src/test/java/com/sequenceiq/consumption/swagger/SwaggerGenerator.java:1]: Wrong package :smile:

File: cloud-consumption/src/test/java/com/sequenceiq/consumption/swagger/SwaggerGenerator.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.ws.rs.ext.ExceptionMapper;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
```
Comment (lajosrodek) [inline cloud-consumption/src/test/java/com/sequenceiq/consumption/swagger/SwaggerGenerator.java:12]: Could you please migrate this to JUnit 5?

File: cloud-consumption/src/test/java/com/sequenceiq/consumption/swagger/SwaggerGenerator.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.sequenceiq.redbeams.api.RedbeamsApi;
+import com.sequenceiq.redbeams.configuration.EndpointConfig;
+import com.sequenceiq.redbeams.configuration.SwaggerConfig;
```
Comment (lajosrodek) [inline cloud-consumption/src/test/java/com/sequenceiq/consumption/swagger/SwaggerGenerator.java:21]: Should be replaced with their consumption counterparts.

File: cloud-consumption/src/test/java/com/sequenceiq/consumption/swagger/SwaggerGenerator.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    @Autowired
+    private SwaggerConfig swaggerConfig;
+
+    @Test
+    public void generateSwaggerJson() throws Exception {
+        Set<Class<?>> classes = new HashSet<>(endpointConfig.getClasses());
+        classes.add(RedbeamsApi.class);
+        Swagger swagger = new Reader(SwaggerConfigLocator.getInstance().getConfig(SwaggerContextService.CONFIG_ID_DEFAULT).configure(new Swagger()))
+                .read(classes);
+        Path path = Paths.get("./build/swagger/redbeams.json");
```
Comment (lajosrodek) [inline cloud-consumption/src/test/java/com/sequenceiq/consumption/swagger/SwaggerGenerator.java:57]: Should be replaced with its consumption counterpart.

**Review by lajosrodek — CHANGES_REQUESTED** [review]

File: cloud-consumption/src/test/java/com/sequenceiq/consumption/swagger/SwaggerGenerator.java (hunk 2 above)
Comment (doktoric) [reply cloud-consumption/src/test/java/com/sequenceiq/consumption/swagger/SwaggerGenerator.java:12]: all of them ? we have such est in every service

Comment (lajosrodek) [reply cloud-consumption/src/test/java/com/sequenceiq/consumption/swagger/SwaggerGenerator.java:12]: I meant this one only. Ideally all existing JUnit 4 tests should be also migrated when making any changes to them, but here it is fine to only consider this new class.

File: scripts/build-dev.sh [hunk 5]
```diff
@@ -11,6 +11,7 @@ if [[ "${RUN_SONARQUBE}" == "true" ]]; then
     ./gradlew -Penv=jenkins -Phttp.socketTimeout=300000 -Phttp.connectionTimeout=300000 -b build.gradle datalake:sonarqube datalake:jacocoTestReport -Dorg.gradle.internal.http.socketTimeout=600000 -Dorg.gradle.internal.http.connectionTimeout=600000 -x test || true
     ./gradlew -Penv=jenkins -Phttp.socketTimeout=300000 -Phttp.connectionTimeout=300000 -b build.gradle environment:sonarqube environment:jacocoTestReport -Dorg.gradle.internal.http.socketTimeout=600000 -Dorg.gradle.internal.http.connectionTimeout=600000 -x test || true
     ./gradlew -Penv=jenkins -Phttp.socketTimeout=300000 -Phttp.connectionTimeout=300000 -b build.gradle redbeams:sonarqube redbeams:jacocoTestReport -Dorg.gradle.internal.http.socketTimeout=600000 -Dorg.gradle.internal.http.connectionTimeout=600000 -x test || true
+    ./gradlew -Penv=jenkins -Phttp.socketTimeout=300000 -Phttp.connectionTimeout=300000 -b build.gradle cloud-consumption:sonarqube redbeams:jacocoTestReport -Dorg.gradle.internal.http.socketTimeout=600000 -Dorg.gradle.internal.http.connectionTimeout=600000 -x test || true
```
Comment (lajosrodek) [inline scripts/build-dev.sh:14]: `redbeams:jacocoTestReport`: This was likely meant to be `cloud-consumption:jacocoTestReport`. :smile:

**Review by lajosrodek — CHANGES_REQUESTED** [review]

**Review by lajosrodek — APPROVED** [review]
Thanks, LGTM

**Pull request merged**